crates/transaction-pool/src/pool/pending.rs

```diff
@@ -243,6 +246,26 @@ impl<T: TransactionOrdering> PendingPool<T> {
         removed
     }
 
+    /// Updates the independent transaction and independent descendants set, assuming the given
+    /// transaction is being _added_ to the pool.
+    fn update_independents(&mut self, tx: &PendingTransaction<T>, tx_id: &TransactionId) {
+        let ancestor_id = tx_id.unchecked_ancestor();
+        if ancestor_id.and_then(|id| self.by_id.get(&id)).is_some() {
+            // the transaction already has an ancestor, so we only need to ensure that the
+            // descendants set includes the highest nonce for this transaction's sender
+            self.independent_descendants.retain(|d| {
+                d.transaction.sender() != tx.transaction.sender() ||
+                    d.transaction.nonce() >= tx.transaction.nonce()
+            });
```

Comment:
hmm, this seems uneccessary, we should be able to simply remove the ancestor and insert the new transaction, right?  

by_id tracks `PendingTransaction` so via ancestor_id we get it and can use that to remove it from `independent_descendants` and then insert the higher nonce